cedar-policy/src/api.rs

```diff
@@ -3384,6 +3384,34 @@ impl Context {
         cedar_policy_validator::context_schema_for_action(&schema.0, action.as_ref())
             .ok_or_else(|| ContextJsonError::missing_action(action.clone()))
     }
+
+    /// Combine two [`Context`]s, returning an error if they contain overlapping keys
+    pub fn merge(self, context: Self) -> Result<Self, ContextCreationError> {
+        Self::from_pairs(self.values().into_iter().chain(context.values()))
+    }
+
+    /// Consume the context and return a `HashMap` of its elements. Note that
+    /// context elements are evaluated when the context is constructed, so the
+    /// stored values may not exactly match the expressions used as input to
+    /// constructors like [`Context::from_pairs`] or [`Context::from_json_str`].
+    pub fn values(self) -> HashMap<String, RestrictedExpression> {
+        self.0
+            .values()
+            .map(|(k, v)| {
+                (
+                    k.to_string(),
+                    match v {
+                        ast::PartialValue::Value(val) => RestrictedExpression(
+                            ast::RestrictedExpr::new_unchecked(ast::Expr::from(val)),
```

Comment:
might be useful to have a `ast::RestrictedExpr::from(val)` function.  From this vantage point it's a little harder to see why `new_unchecked` is safe here